pkg/liquidity-mining/contracts/gauges/SingleRecipientGaugeFactory.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/Clones.sol";
+
+import "./SingleRecipientGauge.sol";
+
+contract LiquidityGaugeFactory {
+    ISingleRecipientLiquidityGauge private _gaugeImplementation;
+
+    mapping(address => bool) private _isGaugeFromFactory;
+
+    event GaugeCreated(address indexed gauge, address indexed recipient);
```

Comment:
What do you think about renaming this to `SingleRecipientGaugeCreated`?

Reply:
Hmm, not sure. I'm guessing the argument in favour is that it avoids us matching the event signature for the `LiquidityGaugeFactory` but with a semantically different address emitted (recipient rather than pool)?

Reply:
Yes, that's what I was thinking. Being explicit in the name also reduces the likelihood of getting them mixed up.

Reply:
Ok, in that case I'm inclined to go whole hog with renaming `LiquidityGaugeV5` to `StakingLiquidityGauge` in that case. It feels weird to have one implementation having the same name with the entire class of contracts.

Reply:
(I'd do this in a separate PR though)